ocaml/typing/typecore.ml

```diff
@@ -3613,19 +3618,16 @@ and type_approx_aux env sexp in_function ty_expected =
   | Pexp_match (_, {pc_rhs=e}::_) -> type_approx_aux env e None ty_expected
   | Pexp_try (e, _) -> type_approx_aux env e None ty_expected
   | Pexp_tuple l ->
-      let tys = List.map
-                  (fun _ -> newvar (Layout.value ~why:Tuple_element)) l
+      let labeled_tys = List.map
+                  (fun (label, _) -> label, newvar (Layout.value ~why:Tuple_element)) l
       in
-      let ty = newty (Ttuple tys) in
+      let ty = newty (Ttuple labeled_tys) in
       begin try unify env ty ty_expected with Unify err ->
         raise(Error(sexp.pexp_loc, env, Expr_type_clash (err, None, None)))
       end;
       List.iter2
-        (function (None, e) -> type_approx_aux env e None
-                | (Some _, _) -> raise(Error(sexp.pexp_loc,
-                                             env,
-                                             Unsupported_labeled_tuple)))
-        l tys
+        (fun (_, e) (_, ty) -> type_approx_aux env e None ty)
+        l labeled_tys
```

Comment:
`type_approx` is used in checking mutually recursive definitions.  We should add some test cases for that (though it might be hard to do much before we have type expressions in which case leave a CR for later).  Also we may need to expand this case when we allow reordering - see the Pexp_record case - leave a CR to revisit then.